Una empresa factura a sus clientes el último día de cada mes. Si el cliente paga
su factura dentro de los primeros 10 días del mes siguiente, tiene un descuento
de $200 o del 2%
de la factura, lo que resulte más conveniente para el cliente.

Si paga en los siguientes 10 días del mes deberá pagar el importe original de la
factura, mientras que si paga después del día 20 deberá abonar una multa de
$350 o del 10% de su factura, lo que resulte mayor. 

Escribir un programa que
lea el número del cliente y el total de la factura, y emita un informe donde conste el número del cliente y los tres importes que podría abonar según la fecha de
pago.

In [18]:
def pago_temprano(importe):
    descuento_2_perc = importe - (importe * 0.02)
    descuento_200 = importe - 200
    return min(descuento_2_perc, descuento_200)

In [19]:
def pago_tardio(importe):
    multa_10_perc = importe + (importe * 0.1)
    multa_350 = importe + 350
    return max(multa_10_perc, multa_350)

In [20]:
def generar_reporte(id_cliente, importe):
    print(f"Si el cliente {id_cliente} paga dentro de los primeros 10 dias, el importe a aboonar es de ${pago_temprano(importe)}")
    print(f"Si el cliente {id_cliente} paga entre los dias 11 y 20 del mes, el importe a aboonar es de ${importe}")
    print(f"Si el cliente {id_cliente} paga despues de los 20 dias, el importe a aboonar es de ${pago_tardio(importe)}")

In [22]:
id_cliente = int(input("Numero de cliente: "))
importe = int(input("Importe de la factura: "))
generar_reporte(id_cliente, importe)

Numero de cliente: 12345
Importe de la factura: 3000
Si el cliente 12345 paga dentro de los primeros 10 dias, el importe a aboonar es de $2800
Si el cliente 12345 paga entre los dias 11 y 20 del mes, el importe a aboonar es de $3000
Si el cliente 12345 paga despues de los 20 dias, el importe a aboonar es de $3350


In [23]:
id_cliente = int(input("Numero de cliente: "))
importe = int(input("Importe de la factura: "))
generar_reporte(id_cliente, importe)

Numero de cliente: 12345
Importe de la factura: 3000000
Si el cliente 12345 paga dentro de los primeros 10 dias, el importe a aboonar es de $2940000.0
Si el cliente 12345 paga entre los dias 11 y 20 del mes, el importe a aboonar es de $3000000
Si el cliente 12345 paga despues de los 20 dias, el importe a aboonar es de $3300000.0


# Ejercicio 1
Se ingresa por teclado el nro de legajo de los alumnos de un curso y su nota de examen final.

El fin de la carga se determina ingresando un -1, en la variable del legajo.

Se debe validar que la nota ingresada sea entre 1 y 10.

Se pide informar:

1) Cantidad de alumnos con nota >= 7

2) Cantidad de alumnos con nota >= 4 y <7

3) Porcentaje de alumnos que están aplazados.

4) Cantidad total de alumnos.

5) La nota más alta con su numero de legajo.

In [ ]:
legajo = int(input("Ingrese un legajo"))
mayor = 0
nota4 = 0
notaint = 0
nota7  = 0
alumno = 0
porc = 0
legajom = -1 # siempre inicializa la variable

while legajo != -1:
    nota=float(input("Ingrese una nota"))
     
    while nota<1 or nota >10:
        print ("Ingrese otra nota entre 1 y 10:")
        nota=float(input("Ingrese una nota"))
    
    if nota<4:
        nota4=nota4+1
    elif nota>=4 and nota<7:
        notaint=notaint+1
    else:
        nota7=nota7+1
   
    if nota > mayor: # si la nota actual es la nota mas alta de las que tenes, 
                    # ahora esta nota es la mas alta, y te guardas el legajo como legajom
                    # (por legajo mayor)
        mayor=nota
        legajom=legajo
    
    # este va indentado aca, porque queres contar TODOS los alumnos,  no solo cuando es mayor la nota
    alumno=alumno +1
    legajo=int(input("Ingrese un legajo"))

if alumno != 0:
    print ("Cantidad de alumnos con nota >= 7 ",nota7)     
    print ("Cantidad de alumnos con nota >= 4 y <7 ",notaint)
    print ("Cantidad de alumnos aplazados ",nota4)
    print ("Cantidad de alumnos",alumno)
    print ("Nota mas alta", mayor, "su nro de legajo es", legajom) 

    porc = nota4 *100/alumno
    print ("Porcentajes de alumnos aplazados ",porc)
else:
    print ("No se ingresaron notas")

# Ejercicio 2
Leer un número entero y mostrar un mensaje informando cuántos dígitos contiene.
Ejemplo: Si se ingresa 12345 debe imprimir 5.

In [44]:
nro = int(input("Ingrese un valor "))

nro_temp = nro # nos quedamos con la copia del numero para el print al final
                # porque le vamos a hacer cambios al numero original

# si el numero es negativo, lo multiplicamos por -1 para sacarle el signo
if nro < 0:
    nro = nro * (-1)

cant = 0 # contador para guardar la cantidad de digitos

while nro > 0: 
    # vamos dividiendo el numero por 10 para sacarle digitos y sumamos en cant para contar cuantas veces
    # lo tuvimos que hacer
    nro = nro//10
    cant = cant + 1

print ("el numero",nro_temp,"tiene",cant,"digitos")

Ingrese un valor 12345
el numero 12345 tiene 5 digitos


ejemplo: -12345

primero le sacamos el signo:

```python
if nro < 0:
    nro = nro * (-1)
```

entonces queda: 12345

cuando entramos en el while:

```python
nro = nro//10
```
nro = 12345 // 10 = 1234 (le sacamos el 5)

sumamos 1 a cant

ahora nro = 1234

nro // 10 = 1234 // 10 = 123 (le sacamos el 4)

sumamos 1 a cant

...